### ISSUE TITLE
block.coinbase can't be used in autoExec()

### ISSUE THREAD

**Issue by luqz** [cryptape/cita#493]
## Description

block.coinbase can't be used in autoExec()

## Steps to Reproduce

1. Add `SysConfig.autoExec=true` when create chain;
2. use `address(block.coinbase).transfer(1)` in function `autoExec()`;
3. create contract and registe it to AutoExec contract;
4. Transfer token to contract.

**Expected behavior**: contract balance sub 1  per block

**Actual behavior**: contract balance remain unchanged

**Reproduce how often**: every time.

## Versions

v0.22.0

## Additional Information

`address(0x5444B13eFe73A7de4d56f2c043424115C8e26eF1).transfer(1)` works but `address(block.coinbase).transfer(1)` not.

**Comment by kaikai1024** [comment]
What's the `Solidity` version?

**Comment by luqz** [comment]
> What's the `Solidity` version?

^0.4.24

**Comment by kaikai1024** [comment]
I've tried the `block.coinbase` on a normal conract and it works.
Maybe something wrong with the auto-exec. 

#483

**Comment by kaikai1024** [comment]
Would you give some other info? @luqz  I can not reproduce.

the contract:

```
pragma solidity ^0.4.24;

contract X {
    address public addr;
    uint public x;

    constructor() payable public {
        addr = block.coinbase;
    }

    function autoExec() public {
        address(block.coinbase).transfer(1);
        x++;
    }
}
```

https://gist.github.com/kaikai1024/74944724904174cd59e4116dd6456b6e

**Comment by luqz** [comment]
Sorry, I try it again.  Your code works well.
I found that maybe the fact is block.coinbase==0x0.
If I remove `require(to != address(0), "Invalid Address")`, It works, but not Expected.
My contract:
`
pragma solidity ^0.4.24;

import './IAutoExec.sol';

contract TestAutoExec is IAutoExec {
    uint public count;

    event Transfer(address indexed from, address indexed to, uint256 indexed value);

    modifier onlyAutoExec {
        if (msg.sender == address(0xFFFfFfFffffffFfFFFFffffFfFFFffFFfF020013))
            _;
        else return;
    }

    function autoExec() external onlyAutoExec {
        incentive();
    }


    function incentive() public {
        _transfer(address(block.coinbase), 1);
        count ++;
    }


    function _transfer(address to, uint value) private returns (bool) {
        require(address(this).balance >= value, "Balance not enough");
        require(to != address(0), "Invalid Address");
        to.transfer(value);
        emit Transfer(address(this), to, value);
    }

    function () public payable {
        
    }

}
`

**Comment by kaikai1024** [comment]
> If I remove require(to != address(0), "Invalid Address") in my contract

Would you please get the coinbase address to double-check.

 > It works, but not Expected.

The balance is not right?

**Comment by luqz** [comment]
> > If I remove require(to != address(0), "Invalid Address") in my contract
> 
> Would you please get the coinbase address to double-check.
> 
> > It works, but not Expected.
> 
> The balance is not right?
I checked the balance of 0x0000000000000000000000000000000000000000,
tokens was transfered to  0x0000000000000000000000000000000000000000。

**Comment by kaikai1024** [comment]
ok, i see. So the problem is why coinbase is zero

**Comment by kaikai1024** [comment]
How do you start the chain? @luqz

**Comment by luqz** [comment]
> How do you start the chain? @luqz

```
./scripts/create_cita_config.py create --super_admin "0x4b5ae4567ad5d9fb92bc9afd6a657e6fa13a2523" --nodes "127.0.0.1:4000,127.0.0.1:4001,127.0.0.1:4002,127.0.0.1:4003" --contract_arguments  SysConfig.economicalModel=1 SysConfig.autoExec=true

./bin/cita bebop setup test-chain/0
./bin/cita bebop start test-chain/0 

./bin/cita bebop setup test-chain/1
./bin/cita bebop start test-chain/1 

./bin/cita bebop setup test-chain/2
./bin/cita bebop start test-chain/2 

./bin/cita bebop setup test-chain/3
./bin/cita bebop start test-chain/3 

```

**Comment by kaikai1024** [comment]
It really can not get the `coinbase` using auto-exec...

**Comment by kaikai1024** [comment]
@luqz It will be fixed at the next version of `v0.23.0`. and we will upgrade the protocol version.

You can use the develop after the #506 be merged.

**Comment by luqz** [comment]
> @luqz It will be fixed at the next version of `v0.23.0`. and we will upgrade the protocol version.
> 
> You can use the develop after the #506 be merged.

Great!👏

**Comment by kaikai1024** [comment]
Reconfirm: not `v0.23.0`. :man_facepalming: 
Check the [new project](https://github.com/cryptape/cita/projects/7)
Check the [new milstone](https://github.com/cryptape/cita/milestone/5)
